### ISSUE TITLE
Sim2sim deploy fails for any sonic_release checkpoint: 8 of 10 encoder observations missing from C++ registry

### ISSUE THREAD

**Issue by Harshita-82** [NVlabs/GR00T-WholeBodyControl#233]
**Summary**

The documented workflow, fine-tune from `sonic_release` → export ONNX → MuJoCo
sim2sim, cannot complete with shipped code.

`policy/release/observation_config_sonic_release.yaml` is the only config producing
the 1751-dim input that sonic_release-derived ONNX models require. It declares 10
encoder observations; **8 are unregistered** in `GetObservationRegistry()`
(`gear_sonic_deploy/src/g1/g1_deploy_onnx_ref/src/g1_deploy_onnx_ref.cpp:1704`).

Each missing observation already exists in the registry under a legacy name. The
Python/config side was renamed; the C++ aliases were never added.

**Steps to reproduce**

Clean clone at aa263a8, TensorRT 10.13.0.35 (the documented version), `just build` clean:

    # 1. Export ONNX from a sonic_release-derived checkpoint
    #    (as documented in user_guide/training.md)
    python gear_sonic/eval_agent_trl.py \
        +checkpoint=<checkpoint>.pt \
        +headless=True ++num_envs=1 \
        +export_onnx_only=true

    # 2. Deploy in sim
    cd gear_sonic_deploy && source scripts/setup_env.sh
    ./deploy.sh --cp <exported>/model_step_XXXXXX \
        --obs-config policy/release/observation_config_sonic_release.yaml \
        sim

**Error**

Everything upstream succeeds, TensorRT compiles the exported encoder without issue:

    INPUT Tensor: obs_dict [1, 1751]. 1751 elements of type FLOAT
    OUTPUT Tensor: encoded_tokens [1, 64]. 64 elements of type FLOAT
    ✓ Encoder initialized successfully!
    ✓ Encoder CUDA graph captured successfully!
    ✓ TensorRT planner model loaded successfully!
    Initializing encoder observations...
    ✗ Error: **Unknown encoder observation function 'encoder_index'!**
    terminate called after throwing an instance of 'std::runtime_error'
      what():  **Invalid encoder observation: unknown function 'encoder_index'
    Aborted (core dumped)**

`encoder_index` is simply the first name the loop rejects; 7 more follow.

## Missing names → existing equivalents

    encoder_index                            4   encoder_mode_4
    command_multi_future_nonflat           580   motion_joint_{positions,velocities}_10frame_step5
    motion_anchor_ori_b_mf_nonflat          60   motion_anchor_orientation_10frame_step5
    command_multi_future_lower_body        240   motion_joint_{pos,vel}_lowerbody_10frame_step5
    vr_3point_local_target                   9   (already registered)
    vr_3point_local_orn_target              12   (already registered)
    motion_anchor_ori_b                      6   motion_anchor_orientation
    smpl_joints_multi_future_local_nonflat 720   smpl_joints_10frame_step1
    smpl_root_ori_b_multi_future            60   smpl_anchor_orientation_10frame_step1
    joint_pos_multi_future_wrist_for_smpl   60   motion_joint_positions_wrists_10frame_step1

Sums to 1750 + 1 leading scalar = 1751, matching the config header and the exported
ONNX input exactly.

**Fix**

PR [#232](https://github.com/NVlabs/GR00T-WholeBodyControl/pull/232) adds alias entries in `GetObservationRegistry()` mapping each new name to its existing gatherer. 

**Environment**

    Commit         aa263a8 (main, 2026-07-31)
    OS             Ubuntu 24.04.4 LTS (Noble), kernel 6.17.0-35-generic
    Python         3.11.15
    GPU            NVIDIA RTX PRO 6000 Blackwell Server Edition (97,887 MiB)
    NVIDIA driver  595.58.03
    CUDA toolkit   12.6
    TensorRT       10.13.0.35
    Isaac Sim      5.1.0.0
    Isaac Lab      2.3.0
    PyTorch        2.7.0+cu128
